enforcer-parent/enforcer/src/main/java/org/wso2/choreo/connect/enforcer/security/jwt/JWTAuthenticator.java

```diff
@@ -435,6 +435,15 @@ private JSONObject validateSubscriptionFromClaim(String name, String version, JW
                     String subTenant = subApi.getAsString(APIConstants.JwtTokenConstants.SUBSCRIBER_TENANT_DOMAIN);
                     if (subTier != null) {
                         validationInfo.setTier(subTier);
+                        if (payload.getClaim(APIConstants.JwtTokenConstants.TIER_INFO) != null) {
+                            JSONObject tierInfo = (JSONObject) payload.getClaim(
+                                    APIConstants.JwtTokenConstants.TIER_INFO);
+                            if (tierInfo.get(subTier) != null) {
+                                JSONObject subTierInfo = (JSONObject) tierInfo.get(subTier);
+                                validationInfo.setStopOnQuotaReach((Boolean)
+                                        subTierInfo.get(APIConstants.JwtTokenConstants.STOP_ON_QUOTA_REACH));
+                            }
+                        }
```

Comment:
Shall we call this as a method defined in KeyValidator?

Reply:
It is only used by this method right?

Reply:
Yes, thought it would be useful for the API key path as well. Had to use a different flow there anyway. We can keep this as it is.